### PR TITLE
Add support for nested classes in source generator

### LINKED ISSUE #3031: [SG] Add support for nested classes
At the moment the SG does not fully support nested classes.

### PR DESCRIPTION
**Pull request by papafe**
Fixes #3031 

##  TODO

* [x] Changelog entry
* [x] Tests (if applicable)

### CODE REVIEW

File: Tests/SourceGenerators/SourceGeneratorAssemblyToProcess/Generated/Realm.SourceGenerator/Realms.SourceGenerator.RealmGenerator/NestedClassWithoutPartialParent.diagnostics.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿[
+  {
+    "Id": "RLM027",
+    "Severity": 3,
+    "Message": "Class OuterClassNonPartial contains nested Realm class OuterClassNonPartial.NestedClassWithoutPartialParent and needs to be declared as partial.",
```
Comment (nirinchev) [inline Tests/SourceGenerators/SourceGeneratorAssemblyToProcess/Generated/Realm.SourceGenerator/Realms.SourceGenerator.RealmGenerator/NestedClassWithoutPartialParent.diagnostics.cs:5]: Now that we have the name of the parent, we probably don't need the full name of the child.

**Review by nirinchev — APPROVED** [review]
Looks reasonable to me

**Review by nirinchev — CHANGES_REQUESTED** [review]
Be sure to check the weaver tests as those failures seem relevant.

**Comment by papafe** [comment]
This should be ready to merge. The CI errors seems to be unrelated

**Review by LaPeste — APPROVED** [review]
It generally makes sense to me.

File: Tests/Realm.Tests/Database/AddOrUpdateTests.cs [hunk 2]
```diff
@@ -1114,110 +1114,111 @@ public void AddMany_WhenSameRefWasDeleted_ShouldThrow()
...
 
-    public partial class NonPrimaryKeyWithNonPKRelation : TestRealmObject
-    {
-        public string StringValue { get; set; }
+        public partial class NonPrimaryKeyWithNonPKRelation : TestRealmObject
+        {
+            public string StringValue { get; set; }
+
+            public NonPrimaryKeyObject OtherObject { get; set; }
+        }
```
Comment (nirinchev) [inline Tests/Realm.Tests/Database/AddOrUpdateTests.cs:1222]: Empty line

**Review by nirinchev — APPROVED** [review]

**Pull request merged**